packages/browser-agent-core/features/jserrors/aggregate/index.js

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { canonicalFunctionName } from './canonical-function-name'
+import { computeStackTrace } from './compute-stack-trace'
+import { stringHashCode } from './string-hash-code'
+import { truncateSize } from './format-stack-trace'
+
+import { registerHandler as register } from '../../../common/event-emitter/register-handler'
+import { HarvestScheduler } from '../../../common/harvest/harvest-scheduler'
+import { stringify } from '../../../common/util/stringify'
+import { handle } from '../../../common/event-emitter/handle'
+import { mapOwn } from '../../../common/util/map-own'
+import { getInfo, getConfigurationValue, getRuntime } from '../../../common/config/config'
+import { now } from '../../../common/timing/now'
+import { cleanURL } from '../../../common/url/clean-url'
+import { FeatureBase } from '../../../common/util/feature-base'
+
+export class Aggregate extends FeatureBase {
+  constructor(agentIdentifier, aggregator) {
+    super(agentIdentifier, aggregator)
+    this.stackReported = {}
+    this.pageviewReported = {}
+    this.errorCache = {}
+    this.currentBody
+
+    this.errorOnPage = false
+
+    // this will need to change to match whatever ee we use in the instrument
+    this.ee.on('interactionSaved',  (interaction) => {
```

Comment:
Could we make this a named class function and move it out of the constructor?

Reply:
agreed